### DISCUSSION TITLE
Armor Stand Meta API Extension

### DISCUSSION THREAD

**Discussion by JakeGBLP** [PaperMC/Paper#10578] [Ideas]
### Suggestion:

Minecraft supports adding EntityData to the armor stand item which will be added to the armor stand once placed down.
It would be really useful to have in [this](https://jd.papermc.io/paper/1.19/com/destroystokyo/paper/inventory/meta/ArmorStandMeta.html) class:

### Solution:

The methods in [this](https://jd.papermc.io/paper/1.19/org/bukkit/entity/ArmorStand.html) class should be added to the [ArmorStandMeta](https://jd.papermc.io/paper/1.19/com/destroystokyo/paper/inventory/meta/ArmorStandMeta.html) class, including the methods to [retrieve and set the armor pieces](https://jd.papermc.io/paper/1.19/org/bukkit/entity/ArmorStand.html#getChestplate()), unless it's preferred to have [another class](https://jd.papermc.io/paper/1.19/org/bukkit/inventory/EntityEquipment.html#getChestplate()) for that.

### Alternatively:

A way to do this without a direct implementation is with NBT, since all of this data is stored in nbt, editing that with accurate values does in fact work, but having methods for them would be better and handier.

**Comment by Machine-Maker** [comment]
Yeah, this has been a big issue with the way the bukkit itemmeta system was designed from the beginning. This will probably be addressed partially in https://github.com/PaperMC/Paper/pull/8711 and the later entity property system. that way support for the whole `EntityData` nbt tag will be added not just for armor stands, but everywhere else it applies like spawn eggs.
